tests/profiles.spec.ts

```diff
@@ -1,23 +1,142 @@
+  await setMemLimit(page, "fixed", "3");
+
+  await deleteProfile(page, profile);
+});
+
+test("profile resource limits", async ({ page }) => {
+  const profile = randomProfileName();
+  await createProfile(page, profile);
+  await editProfile(page, profile);
+
+  await page.getByText("Resource limits").click();
+  await setOption(page, "Memory swap", "true");
+  await setOption(page, "Disk priority", "1");
+  await setInput(page, "Max number of processes", "Enter number", "2");
+  await saveProfile(page);
+
+  await assertReadMode(page, "Memory swap (Containers only)	Allow");
+  await assertReadMode(page, "Disk priority 1");
+  await assertReadMode(page, "Max number of processes (Containers only) 2");
+
+  await deleteProfile(page, profile);
+});
+
+test("profile security policies", async ({ page }) => {
+  const profile = randomProfileName();
+  await createProfile(page, profile);
+  await editProfile(page, profile);
+  await page.getByText("Security policies").click();
+
+  await setOption(page, "Prevent the instance from being deleted", "true");
+  await setOption(page, "Run the instance in privileged mode", "true");
+  await setOption(
+    page,
+    "Prevent instance file system from being UID/GID shifted",
+    "true"
+  );
+  await setInput(page, "Base host id", "Enter ID", "11");
+  await setInput(page, "Idmap size", "Enter number", "22");
+  await setOption(page, "Unique idmap usage", "true");
+  await setOption(page, "Allow /dev/lxd in the instance", "true");
+  await setOption(page, "Make /1.0/images API available", "true");
+  await setOption(page, "Enable secureboot", "true");
+  await saveProfile(page);
+
+  await assertReadMode(page, "Prevent the instance from being deleted Yes");
+  await assertReadMode(
+    page,
+    "Run the instance in privileged mode (Containers only) Allow"
+  );
+  await assertReadMode(
+    page,
+    "Prevent instance file system from being UID/GID shifted on startup (Containers only) Yes"
+  );
+  await assertReadMode(page, "Base host id (Containers only) 11");
+  await assertReadMode(page, "Idmap size (Containers only) 22");
+  await assertReadMode(page, "Unique idmap usage (Containers only) Yes");
+  await assertReadMode(
+    page,
+    "Allow /dev/lxd in the instance (Containers only) Yes"
+  );
+  await assertReadMode(
+    page,
+    "Make /1.0/images API available over /dev/lxd (Containers only) Yes"
+  );
+  await assertReadMode(page, "Enable secureboot (VMs only) true");
+
+  await deleteProfile(page, profile);
+});
+
+test("profile snapshots", async ({ page }) => {
+  const profile = randomProfileName();
+  await createProfile(page, profile);
+  await editProfile(page, profile);
+  await page.getByText("Snapshots").click();
+
+  await setInput(page, "Snapshot name", "Enter name pattern", "snap123");
+  await setInput(page, "Expire after", "Enter expiry expression", "3m");
+  await setOption(page, "Snapshot stopped instances", "true");
+  await saveProfile(page);
```

Comment:
It is a bit complicated because the Schedule keyword is already used in the help text for the Snapshot name pattern. I'll add a custom function for the schedule.